app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,3 @@
+class Users::SessionsController < Devise::SessionsController
+  after_action :verify_authorized, except: [:new, :create, :destroy]
```

Comment:
@NickyBobby Checking for teacher in the view is separate from this. If you assume that all actions need to be authorized unless explicitly stated, I believe that this is saying that you don't need to be authorized to login or view the root page which has a link to login.